UI/UISearchBar.cs

```diff
@@ -107,8 +107,17 @@ private void HandleTextInput()
 				prev = prev.Remove(cursorPosition);
 
 			string newString = Main.GetInputText(prev);
-			if ((Main.keyState.IsKeyDown(Keys.LeftControl) || Main.keyState.IsKeyDown(Keys.RightControl)) && KeyTyped(Keys.Back))
-				DeleteWord(ref newString);
+
+
+			if (Main.keyState.IsKeyDown(Keys.LeftControl) || Main.keyState.IsKeyDown(Keys.RightControl))
+			{
+				if (KeyTyped(Keys.Back))
+					DeleteWord(ref newString);
+				else if (KeyTyped(Keys.Left))
+					HandleLeft();
+				else if (KeyTyped(Keys.Right))
+					HandleRight();
+			}
```

Comment:
I still feel the cursor movement should be handled down with the other cursor movement stuff. If you're that worried about having to again check if Ctrl is pressed, just cache the result in a var

Reply:
You are probably right. I just changed it, and it looks better that way. I will commit later.